app/services/metrics/base_metric_processor.rb

```diff
@@ -0,0 +1,84 @@
+module Metrics
+  ##
+  # Base class for classes that do the processing of a Metric from a
+  # collection of events.
+  class BaseMetricProcessor < BaseService
+    def initialize(events:, time_interval:)
```

Comment:
suggestion: use attr readers to avoid the `@` all over the place

Reply:
and as a general comment all over the file: use the attr_readers instead of passing the same params all over the place